core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/resourcePolicies/ResourceAllocation.scala

```diff
@@ -0,0 +1,40 @@
+package edu.uci.ics.amber.engine.architecture.scheduling.resourcePolicies
+
+import edu.uci.ics.amber.engine.architecture.scheduling.Region
+import edu.uci.ics.amber.engine.common.AmberConfig
+import edu.uci.ics.texera.workflow.common.workflow.PhysicalPlan
+
+import scala.collection.mutable.ListBuffer
+
+object ResourceAllocation {
+  def apply(region: Region): ResourceAllocation = {
+    ResourceAllocation(region)
+  }
+}
+
+case class ResourceAllocation(region: Region) {
+
+  private def physicalOpIds = region.physicalOpIds.filter(_.logicalOpId.id.contains("PythonUDF"))
```

Comment:
1. please name it as `BasicResourceAllocator`.
2. please remove `region` from its constructor. There should be one Allcaotor for all regions. Instead, one physical plan can have one ResourceAllocator, so you may add the physical plan to the constructor. Alternatively you can pass nothing in the constructor and pass all necessary parameters through the allocate function call.